src/Lib/Util/StringUtil.hpp

```diff
@@ -21,8 +21,14 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #include "Core/CoreTypes.hpp"
 #include <vector>
 
+#define sn_check(a, b) real_sn_check(a, b, __func__, __LINE__)
```

Comment:
You want me to proceed fixing a few more warnings using this macro like I have it? Or something a little different? Any reason why we're not using Cpp string methods in place of sprintf?

Reply:
@andy5995 the ideal fix here is to just use cpp string stuff IMO.

Reply:
ok, converted some things. On the right track now?